src/Encoder.ts

```diff
@@ -144,8 +157,8 @@ export class Encoder<ContextType = undefined> {
       }
     } else if (typeof object === "string") {
       this.encodeString(object);
-    } else if (this.useBigInt64 && typeof object === "bigint") {
-      this.encodeBigInt64(object);
+    } else if (this.forceBigIntToInt64 && typeof object === "bigint") {
```

Comment:
Here's my proposal for encoding options: let's keep `useBigInt64`, but make it default to true. It would only be used here (i.e. `} else if (this.useBigInt64 && typeof object === "bigint") {`).

The only reason to ever disable it is if you have a custom extension for encoding bigints from a legacy msgp version.

That way we can remove the changes to `encodeObject` and the route for encoding a bigint is simpler.

Reply:
I still think it's useful to allow `forceBigIntToInt64` option since you then have an ability to ensure a fixed length encoding by using bigints. It feels like a really useful mechanism. I've simplified the logic based on your suggestion though, but kept backwards compatibility with custom extensions because it didn't negatively impact things.